server/controllers/chapterController.ts

```diff
@@ -26,30 +26,46 @@ export default {
       description,
       category,
       details,
-      location_id,
-      creator_id,
+      location,
+      creator,
     } = req.body;
 
     const chapter = new Chapter({
       name,
       description,
       category,
       details,
-      location_id,
-      creator_id,
+      location,
+      creator,
     });
 
     try {
       await chapter.save();
       res.status(201).json(chapter);
     } catch (e) {
+      if (e.code === '23503' && e.message.includes('foreign key constraint')) {
+        if (e.detail.includes('location')) {
```

Comment:
Just an edge case, if e.detail is **null** or **undefined** than it will throw an error like ".includes of undefined". To prevent this we can use destructing with default value.